src/index.ts

```diff
@@ -0,0 +1,181 @@
+// global
+import { FastifyPluginAsync } from "fastify";
+import { Item, Member, IdParam, Actor } from "graasp";
+
+// local
+import { CreateActionTask } from './services/action/create-action-task';
+import { DeleteActionsTask } from './services/action/delete-actions-task';
+import { GetItemTask } from './services/item/get-item-task';
+import { BaseAction } from './services/action/base-action';
+import { ActionService } from './db-service';
+import { Action } from "./interfaces/action";
+import { paths } from './constants/constants';
+import { correctStatusCodes } from './constants/constants';
+
+import {
+  getOne
+} from './schemas/schemas';
+
+
+export interface GraaspActionsOptions {
+  graaspActor: Actor;
+}
+
+const actionService = new ActionService();
+
+const plugin: FastifyPluginAsync<GraaspActionsOptions> = async (fastify, options) => {
+  const { graaspActor } = options;
+  const {
+    items: { taskManager, dbService: iS },
+    members: {dbService: memberService},
+    itemMemberships: {dbService: itemMembershipsService},
+    taskRunner: runner,
+    log: defaultLogger,
+  } = fastify;
+
+  // save action when a item is created
+  const createItemTaskName = taskManager.getCreateTaskName();
+  runner.setTaskPostHookHandler(createItemTaskName, async (item: Partial<Item>, actor, { log, handler }) => {
+    const member = actor as Member;
+    const extra = {memberId: actor.id, itemId: item.id};
+    const view = 'builder';
+    const action: Action = new BaseAction(actor.id, item.id, member.type, item.type, "create", view, extra);
+    actionService.create(action, handler);
+  });
+  
+  // delete all the actions matching the given `memberId`
+  fastify.delete<{ Params: IdParam }>(
+    '/members/:id/delete-actions',
+    { schema: getOne },
+    async ({ member, params: { id }, log,  }, reply) => {
+      const task = new DeleteActionsTask(member, id, actionService);
+      await runner.runSingle(task);
+
+      reply
+      .code(200)
+      .header('Content-Type', 'application/json; charset=utf-8')
+      .send( task.result )      
+    },
+  );
+
+};
+
+
+// save action (except create and delete) for the itemId with its actionType and view
+const createActionTask = async function (member: Member, itemId: string, actionType: string, view: string, runner, itemService) {
+  // get item
+  const t1 = new GetItemTask(member, itemId, itemService);
```

Comment:
I don't know the exact syntax, but I think you can reuse the task in graasp core. You can use `taskManager.createGetTask` (defined in graasp core item task-manager). So you don't need to create your own get item! Same goes for member.

Reply:
Changed. Now it is used the `taskManager.createGetTask` when retrieving an item. It is removed all the item tasks.